shell.nix

```diff
@@ -1,14 +1,30 @@
 {
   pkgs ? import <nixpkgs> { },
-  packages ? [ ],
+  treefmt ? null,
+  nix-direnv ? (pkgs.callPackage ./default.nix { }),
+  test_pkgs ? (pkgs.lib.callPackagesWith pkgs ./tests { inherit nix-direnv; }),
 }:
-
-with pkgs;
-mkShell {
-  packages = packages ++ [
-    python3.pkgs.pytest
-    python3.pkgs.mypy
-    ruff
-    direnv
-  ];
+let
+  inherit (pkgs) lib;
+in
+pkgs.mkShell {
+  DIRENV_STDLIB = "${test_pkgs.direnv-stdlib}";
+  DIRENVRC = "${nix-direnv}/share/nix-direnv/direnvrc";
+  BATS_LIB_PATH = lib.strings.makeSearchPath "" (
+    with test_pkgs;
+    [
+      bats-support
+      bats-assert
+    ]
+  );
+  packages =
+    (builtins.attrValues {
+      inherit (pkgs)
+        bats
+        direnv
+        shellcheck
+        ;
+    })
+    ++ (builtins.attrValues (lib.attrsets.filterAttrs (name: _val: name != "direnv-stdlib") test_pkgs))
```

Comment:
Because `direnv-stdlib` is the result of `fetchurl`, it's a single-file in the store. This causes the symlinking to create the package environment to abort hard on it. We just filter it here so it is available when setting `DIRENV_STDLIB` but not in the package set.